examples/spells/rareSpells.qnt

```diff
@@ -23,4 +23,30 @@ module rareSpells {
     assert(2.to(5).mapBy(i => i * 2).mapValuesSum() == 28),
     assert(Map(2 -> -4, 4 -> 2).mapValuesSum() == -2),
   }
-}
\ No newline at end of file
+
+  //// Returns a list of all elements of a set.
+  //// The ordering will be arbitrary.
+  ////
+  //// - @param __set a set
+  //// - @returns a list of all elements of __set
+  pure def toList(__set: Set[a]): List[a] = {
```

Comment:
IMO this is a problem inherited from `fold`, and `toList` is a fine name given that `fold` is not called `foldSomehow` ;)

If we want to be more principled about ordering we could make `fold` take a comparitor op or require quint sets be constructed over a totally ordered domain?

More worrying IMO is that, according to the specification of `fold` this operator technically has "undefined behavior":

https://github.com/informalsystems/quint/blob/f5828700e3530d73a75e8fe5f710ed79eaa60033/doc/builtin.md#L202-L203)

An alternative here would be provide this as `toList(s: Set[a], cmp: (a, a) => int): List[a]` where `cmp` is an ordering relation determining the way the list is to be constructed.

Reply:
> An alternative here would be provide this as toList(s: Set[a], cmp: (a, a) => int): List[a] where cmp is an ordering relation determining the way the list is to be constructed.

This sounds like the only good alternative to me. In this case, I would call the operator `toSortedList`.

Reply:
> > require quint sets be constructed over a totally ordered domain
> 
> That idea of mine wouldn't work due to -- if nothing else -- sets of functions.

Well, this is one of the challenges when dealing with as expressive languages as TLA+. I can imagine how we could order functions (that is, maps). For instance, we could write them as an ordered set of key-value pairs. I am not saying that it would be efficient, but doable (in theory).

Reply:
I've made an update to include the comparison operator as an argument of `toList`.